test/objectid.test.js

```diff
@@ -32,41 +34,75 @@ describe('ObjectID', function() {
   });
 
   it('should convert 24 byte hex string as ObjectID', function() {
-    var ObjectID = db.connector.getDefaultIdType();
-    var str = '52fcef5c0325ace8dcb7a0bd';
-    ObjectID(str).should.be.an.instanceOf(db.ObjectID);
+    var ObjectID = ds.connector.getDefaultIdType();
+    var str = objectIDLikeString;
+    ObjectID(str).should.be.an.instanceOf(ds.ObjectID);
   });
 
   it('should not convert 12 byte string as ObjectID', function() {
-    var ObjectID = db.connector.getDefaultIdType();
+    var ObjectID = ds.connector.getDefaultIdType();
     var str = 'line-by-line';
     ObjectID(str).should.be.equal(str);
   });
 
   it('should keep mongodb ObjectID as is', function() {
-    var ObjectID = db.connector.getDefaultIdType();
-    var id = new db.ObjectID();
-    ObjectID(id).should.be.an.instanceOf(db.ObjectID);
+    var ObjectID = ds.connector.getDefaultIdType();
+    var id = new ds.ObjectID();
+    ObjectID(id).should.be.an.instanceOf(ds.ObjectID);
   });
 
   it('should keep non-string id as it', function() {
-    var ObjectID = db.connector.getDefaultIdType();
+    var ObjectID = ds.connector.getDefaultIdType();
     var id = 123;
     ObjectID(id).should.be.equal(123);
   });
 
-  it('coerces ObjectID', function() {
-    const coercedId = db.connector.isObjectIDProperty('Book', {}, '52fcef5c0325ace8dcb7a0bd');
-    coercedId.should.be.True();
-  });
+  context('strictObjectIDCoercion', function () {
+    context('set to false (default)', function () {
+
+      beforeEach(function(done) {
+        Book.deleteAll(done);
+      });
+
+      it('it should coerce to ObjectID', async function() {
+        Book = ds.createModel(
+          'book',
+          {
+            xid: {type: String}
+          }
+        );
+        const book = await Book.create({xid: objectIDLikeString});
+        book.xid.should.be.an.instanceOf(ds.ObjectID);
+      });
+    });
+
+    context('set to true', function () {
+      it('should not coerce to ObjectID', async function() {
+        Book = ds.createModel(
+          'book',
+          {
+            xid: {type: String}
+          },
+          {strictObjectIDCoercion: true}
+        );
+        const book = await Book.create({xid: objectIDLikeString});
+        book.xid.should.equal(objectIDLikeString);
+      });
+
+      it('if type is set to ObjectID, should coerce to ObjectID', async function() {
+        Book = ds.createModel(
+          'book',
+          {
+            xid: {type: String, mongodb: {dataType: ObjectID}}
+          },
+          {strictObjectIDCoercion: true}
+        );
+        const book = await Book.create({xid: objectIDLikeString});
+        book.xid.should.be.an.instanceOf(ds.ObjectID);
```

Comment:
@b-admike any idea why this fails despite the line 2095 being executed in `/lib/mongodb.js`?